### DISCUSSION TITLE
fund function giving error

### DISCUSSION THREAD

**Discussion by Lucian-Ryu** [Cyfrin/foundry-full-course-cu#243] [General]
hello, im testing the FundMe.sol contract on remix and when i try using the fund function is giving me an error where the transaction dosnt go trough. it says the function should be payable, but it is.

am i missing something here?

here is my code 

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {PriceConverter} from  "./PriceConverter.sol";

contract FundMe{

    using PriceConverter for uint256;
    
    uint256 public minimumUsd  = 5e18;

    address[] public funders;
    mapping(address funder => uint256 amountFunded) public addressToAmountFunded;

    address public owner;

    constructor() {
        owner = msg.sender;
    }

    modifier onlyOwner() {
        require(msg.sender == owner, "Sender is not owner");
        _;
    }

    function fund() public payable {
       require (msg.value.getConversionRate() >= minimumUsd, "didnt send enough ETH");  //1e18 = 1 ETH
       funders.push(msg.sender);
       addressToAmountFunded[msg.sender] +=  msg.value;
    }

   function withdraw() public onlyOwner {
       for(uint256 funderIndex = 0; funderIndex < funders.length; funderIndex++){
           address funder = funders[funderIndex];
           addressToAmountFunded[funder] = 0;
       }
       funders = new address[](0);
       (bool callSuccess, ) = payable(msg.sender).call{value: address(this).balance}("");
       require(callSuccess, "Call failed");

   }
    
}

**Comment by dimitar-dermendzhiev** [comment]
I would guess you are trying to call the function on the wrong environment. If you are trying to call the fund function on Remix VM then you  can't fetch the price of ETH. If I am not mistaken your transaction can't get through because whatever value you enter the getConversionRate will always return 0, so you have something like  0 >= 5e18. Change your environment to Injected Provider - Metamask

**Comment by PatrickAlphaC** [comment]
Hello! Thanks for making this issue. 

Could you please remember to follow this section to format your questions? https://youtu.be/umepbfKp5rI?t=22464
